File: python3/pascals_triangle.py
from typing import List

"""
Space   : O(n^2)
Time    : O(n^2)
"""

class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        ans = []

        for i in range(1, numRows+1):
            row = []
            for j in range(i):
                if j == 0 or j == i-1:
                    row.append(1)
                else:
                    row.append(ans[-1][j-1] + ans[-1][j])
            ans.append(row)

        return ans

File: golang/pascals_triangle.go
package main

// Space	: O(n**2)
// Time		: O(n**2)

func generate(numRows int) [][]int {
    var ans [][]int
    
    for i := 1; i <= numRows; i++ {
        var row []int
        for j := 1; j <= i; j++ {
            if j == 1 || j == i {
                row = append(row, 1)
            } else {
                row = append(row, ans[i-2][j-2] + ans[i-2][j-1])
            }
        }
        ans = append(ans, row)
    }
    
    return ans
}